libminifi/test/archive-tests/CompressContentTests.cpp

```diff
@@ -91,851 +91,835 @@ class ReadCallback: public org::apache::nifi::minifi::InputStreamCallback {
+    tempDir_ = get_global_controller().createTempDirectory(format);
+    REQUIRE(!tempDir_.empty());
+    raw_content_path_ = utils::file::FileUtils::concat_path(tempDir_, "minifi-expect-compresscontent.txt");
+    compressed_content_path_ = utils::file::FileUtils::concat_path(tempDir_, "minifi-compresscontent");
+    initContentWithRandomData();
+  }
+
+  template<class ...Args>
+  void writeCompressed(Args&& ...args){
+    std::ofstream file(compressed_content_path_, std::ios::binary);
+    file.write(std::forward<Args>(args)...);
+  }
+};
+
+class DecompressTestController : public CompressDecompressionTestController{
+ public:
+  ~DecompressTestController(){
+    tempDir_ = "";
+    raw_content_path_ = "";
+    compressed_content_path_ = "";
   }
+};
+
+TEST_CASE("CompressFileGZip", "[compressfiletest1]") {
+  CompressTestController testController;
+
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::processors::CompressContent>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::processors::LogAttribute>();
+  LogTestController::getInstance().setTrace<core::ProcessSession>();
+  LogTestController::getInstance().setTrace<core::ProcessContext>();
+  LogTestController::getInstance().setTrace<core::repository::VolatileContentRepository>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::Connection>();
+  LogTestController::getInstance().setTrace<org::apache::nifi::minifi::core::Connectable>();
+
+  std::shared_ptr<TestRepository> repo = std::make_shared<TestRepository>();
+
+  std::shared_ptr<core::Processor> processor = std::make_shared<org::apache::nifi::minifi::processors::CompressContent>("compresscontent");
+  std::shared_ptr<core::Processor> logAttributeProcessor = std::make_shared<org::apache::nifi::minifi::processors::LogAttribute>("logattribute");
+  processor->initialize();
+  utils::Identifier processoruuid;
+  REQUIRE(true == processor->getUUID(processoruuid));
+  utils::Identifier logAttributeuuid;
+  REQUIRE(true == logAttributeProcessor->getUUID(logAttributeuuid));
+
+  std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::VolatileContentRepository>();
+  // std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::FileSystemRepository>();
+
+  content_repo->initialize(std::make_shared<org::apache::nifi::minifi::Configure>());
+  // connection from compress processor to log attribute
+  std::shared_ptr<minifi::Connection> connection = std::make_shared<minifi::Connection>(repo, content_repo, "logattributeconnection");
+  connection->addRelationship(core::Relationship("success", "compress successful output"));
+  connection->setSource(processor);
+  connection->setDestination(logAttributeProcessor);
+  connection->setSourceUUID(processoruuid);
+  connection->setDestinationUUID(logAttributeuuid);
+  processor->addConnection(connection);
+  // connection to compress processor
```

Comment:
Wouldn't using a `TestPlan` simplify most of this anyway?

Reply:
`TestPlan` does not seem appropriate since this tests a single processor, and it uses raw connections